src/source.cc

```diff
@@ -88,9 +88,83 @@ std::string Source::FixIt::string(Glib::RefPtr<Gtk::TextBuffer> buffer) {
 const std::regex Source::View::bracket_regex("^([ \\t]*).*\\{ *$");
 const std::regex Source::View::no_bracket_statement_regex("^([ \\t]*)(if|for|else if|while) *\\(.*[^;}] *$");
 const std::regex Source::View::no_bracket_no_para_statement_regex("^([ \\t]*)(else) *$");
+const Glib::ustring right_click_menu_xml= R"RAW(<interface>
+    <menu id='right-click-line-menu'>
+      <section>
+        <item>
+          <attribute name='label' translatable='yes'>_Undo</attribute>
+          <attribute name='action'>app.edit_undo</attribute>
+        </item>
+        <item>
+          <attribute name='label' translatable='yes'>_Redo</attribute>
+          <attribute name='action'>app.edit_redo</attribute>
+        </item>
+      </section>
+      <section>
+        <item>
+          <attribute name='label' translatable='yes'>_Paste</attribute>
+          <attribute name='action'>app.edit_paste</attribute>
+        </item>
+      </section>
+      <section>
+        <item>
+          <attribute name='label' translatable='yes'>_Toggle _Line _Comment</attribute>
+          <attribute name='action'>app.source_comments_toggle</attribute>
+        </item>
+        <item>
+          <attribute name='label' translatable='yes'>_Rename</attribute>
+          <attribute name='action'>app.rename</attribute>
```

Comment:
In order to make it work, I had to change `app.rename` to `app.source_rename` (same as in menu.cc).